packages/fetchai/connections/http_server/connection.py

```diff
@@ -166,22 +166,19 @@ async def create(cls, http_request: BaseRequest) -> "Request":
         return request
 
     def to_envelope_and_set_id(
-        self, connection_id: PublicId, agent_address: str, dialogues: HttpDialogues,
+        self, dialogues: HttpDialogues, target_skill_id: PublicId,
     ) -> Envelope:
         """
         Process incoming API request by packaging into Envelope and sending it in-queue.
 
-        :param connection_id: id of the connection
-        :param agent_address: agent's address
         :param dialogue_reference: new dialog refernece for envelope
+        :param target_skill_id: the target skill id
 
         :return: envelope
         """
         url = self.full_url_pattern
-        uri = URI(self.full_url_pattern)
-        context = EnvelopeContext(connection_id=connection_id, uri=uri)
```

Comment:
This was somewhat problematic. Firstly, an external entity should not be able to target a skill. Second, if we don't know the target skill we cannot maintain a proper bilateral interaction as we don't know the address (using agent address is not ok as the HTTPDialogues class in the skill will be instantiated with the skill_id as an address - cf. component to component messaging)